src/lib/core/popen.c

```diff
@@ -1178,7 +1178,13 @@ popen_new(struct popen_opts *opts)
 	 * vfork to complete. Also we try to do as minimum
 	 * operations before the exec() as possible.
 	 */
+#pragma GCC diagnostic push
+	/*
+	 * FIXME: we need to review popen's design and refactor this part.
+	 */
```

Comment:
Nit: please file a ticket with a proper description and mention it here for the reference:

```C
/* TODO(gh/XXXX): Rework popen so that it doesn't need to fork the main process. */
```

Reply:
Fixed: created a new issue regarding `vfork` in `popen` and mentioned it in the TODO.